src/transformers/modeling_utils.py

```diff
@@ -227,6 +227,7 @@ def invert_attention_mask(self, encoder_attention_mask: Tensor) -> Tensor:
         # /transformer/transformer_layers.py#L270
         # encoder_extended_attention_mask = (encoder_extended_attention_mask ==
         # encoder_extended_attention_mask.transpose(-1, -2))
+        ###TODO
         encoder_extended_attention_mask = encoder_extended_attention_mask.to(dtype=self.dtype)  # fp16 compatibility
```

Comment:
@stas00 is `self.dtype` supposed to be 16-bit when using amp?  My own print statements are saying float32

Reply:
autocast keeps everything in the original dtype (usually fp32) and only does ops that are supported in half precision when they are called. It keeps a cache of the half-precision version of the original weights for the scope of autocast. 

Also `self.dtype` in the context you highlighted only checks the first param.

Deepspeed does it differently. it keeps weights in fp16 and uses fp32 master weights only when needed. So that's why we have those checks in the code as we have "torch.float16" then - also when we use `--fp16_full_eval`